### DISCUSSION TITLE
Not a fan of the new extensions interface

### DISCUSSION THREAD

**Discussion by gruevy** [vladmandic/sdnext#990] [General]
All I want it to show me, is if there's and update or not. None of these things are particularly useful to me:

Created Sun Apr09 2023 08:20 | Added Mon Apr10 2023 00:00 | Pushed Tue May09 2023 12:30 | Updated Tue May09 2023 12:30
Stars 234 | Size 117 | Commits 34 | Issues 5

They're just visual spam cluttering the page.

I also don't care to be shown that the current version is "e474a308 Mon May15 2023 09:29" unless you're also going to tell me whether it's the latest one and if there's an update I should have. The update button is there whether I have an update or not, as far as I can tell, which has resulted in me clicking and restarting the server several times wondering if anything was changing and never actually knowing.

The old interface was better. I like most of the changes you've been making but not this one. That's my $0.02. Thanks for all the hard work.

**Comment by vladmandic** [comment]
> They're just visual spam cluttering the page.

we can agree to disagree.

> The update button is there whether I have an update or not

that is not the intention. update button should only show up if actual updates are available.
make sure you're on latest version and copy&paste the line where its showing up incorrectly.

> The old interface was better. I like most of the changes you've been making but not this one.

be forward looking. if there is something that can be improved about the interface, let me know. saying "old was better" is not helping me.

**Reply by gruevy** [reply]
> be forward looking. if there is something that can be improved about the interface, let me know. saying "old was better" is not helping me.

Fair enough. If a user goes to the extensions tab, they're looking to do one of two things: update extensions, or install/uninstall one. Considering that the main things you want to do should be the most prominent and easiest to find, here's what I'd change. 

-The user-installed extensions should appear at the top of the list instead of the built-in ones, because those are ones I actually want to manipulate.
-The update button should move to the left to be the second column. Name, then update, since that's what I'm doing there. All the clutter can move to the side where it can be ignored.
-The update button should be renamed to say "update available" and the box it's in might have additional info, like "new version downloaded, pending restart" or something else that'll tell me what's going on. Just saying "updating" after you click it doesn't tell me what is happening or what I need to do.
-The uninstalled (new) extensions should move from the bottom of the list back to their own tab. If I want a new extension, I don't want to scroll past what I already have first, and if I don't want a new extension, there's no reason to load the new extensions list.

Now, if you want to get crazy with it, a tiles-based interface for extensions that includes useful expandable screenshots would be awesome. It'd show me what I'm actually getting.

**Reply by gruevy** [reply]
Oh, forgot to mention: The two offenders that don't update and the update button always stays visible are:

https://github.com/ashen-sensored/sd_webui_SAG
https://github.com/bbc-mc/sdweb-clip-changer

Might be a problem with just those two, since they were both manual installs.

**Reply by vladmandic** [reply]
regarding update available - just to make sure, are you on latest version of the app? because there was an issue with earlier versions where it was comparing currently installed to "any code" published on extension, even if it was not comitted to main branch.
and after updating to latest, make sure you click refresh list on top right so it pulls latest metedata.

the rest of feedback is pretty much how you'd like to have a split view - installed vs available.

**Reply by gruevy** [reply]
> regarding update available - just to make sure, are you on latest version of the app

I always run 'git pull' before I start it up. I just doublechecked and the version numbers match, so I think so.

If I had to pick between a split view and the other changes I mentioned, I'd go with the other changes and not split the view.

**Reply by vladmandic** [reply]
a lot of other changes you mentioned are about sort order - and you can pick that already.
if there is a specific sort/order that's missing and you'd like to have, let me know.
what's left is moving columns left/right.

**Reply by gruevy** [reply]
I didn't see a sort order for 'my additional extensions first'. And now that I think about it, 'update available' would be a nice option too.

All of this is just my feedback and it might not be a good use of your time, so no worries if you decide not to mess with it.

**Reply by vladmandic** [reply]
done.

**Reply by gruevy** [reply]
Works, thanks!

**Comment by vis223** [comment]
As far as I can tell I was not able to sort extensions so the newest releases showed at the top. When sorted by date the oldest appeared first and I could see no way to reverse the order. Apologies if I was being dense but it was always the way I checked for new and interesting extensions. Did I miss something in the new interface?

**Reply by vladmandic** [reply]
if i sort by "updated date", i get newest first. 
just make sure you click "refresh" before to get latest list.

**Comment by vis223** [comment]
When I follow those instructions the top of the list shows as 'batch face swap' and 'model keyword.' 
These may be the most recent updates but they are definitely not the most recent releases.
I scroll down quite a bit before I see an extension I don't recognize and want to investigate.

Is there a way to add an asc/desc method to sort by date?

I'd love a way to quickly check if something new has come out

**Reply by vladmandic** [reply]
i'll change the default order of "created date" so it shows newest extensions first.

**Comment by vis223** [comment]
And while I have your attention, if you don't mind...
The brilliant 'Enable Dynamic Thresholding (CFG Scale Fix)' has a 'view the wiki' link that is really handy.
Its probably  a lot of work but why can't all the scripts gave a quick link to their git hub or readme?
If I want a quick reminder on syntax using x/y/z or any other script right now the quickest way is to open the extension tab, search  for it then click on its title to go to their github. 
It's not ideal and I know it should be the developers implementing it themselves but since you already have these links working within the extension tab I wondered if it would be possible.

Great work on the FAQ's by the way. It's a really helpful format. I wish I was more knowledgeable to lend you a hand.
You've a lot on your plate so I hope others volunteer to help you soon. You are doing such a good job in all areas.
It's very much appreciated

**Reply by vladmandic** [reply]
its a valid idea, but not sure if app should draw the link in extensions ui. the problem is that app doesn't even know if script is using fixed ui elemements, accordion element or tabs - it would have to account for all border cases and it can lead to a lot of ui issues.

but like you said, i'd love when each extension author added that link themselves.
